mypy/nodes.py

```diff
@@ -2660,8 +2679,10 @@ def fullname(self) -> Optional[str]:
     @property
     def type(self) -> 'Optional[mypy.types.Type]':
         node = self.node
-        if ((isinstance(node, Var) or isinstance(node, FuncBase))
-                and node.type is not None):
+        if (isinstance(node, Var) and node.type is not None):
+            return node.type
+        # mypy thinks this branch is unreachable but it is wrong
+        elif (isinstance(node, FuncBase) and node.type is not None):
```

Comment:
Hm, is this a bug then? Do we have an issue for this?

Reply:
It's the "intersection types bug". #3603. I'll point to it in the comments